fortitude/src/rules/correctness/conditionals.rs

```diff
@@ -0,0 +1,134 @@
+            // Comments on the same line are permitted.
+            let mut current_node = semicolon_node;
+            while let Some(next) = current_node.next_sibling() {
+                if next.kind() == ";" {
+                    current_node = next;
+                    continue;
+                }
+                if next.start_position().row != semicolon_node.start_position().row
+                    || next.kind() == "comment"
+                {
+                    return None;
+                }
+                break;
+            }
+            let if_edit = node.edit_replacement(src, ifthenify(node, src).ok()?);
+            let indentation = node.indentation(src);
+            // To replace the semicolon node, we should also replace all
+            // trailing whitespace.
+            let semicolon_start = semicolon_node.start_textsize();
+            let mut semicolon_end = semicolon_node.end_byte();
+            let text = src.source_text().as_bytes();
+            while text[semicolon_end] == b' ' || text[semicolon_end] == b'\t' {
+                semicolon_end += 1;
+            }
+            let semicolon_end = TextSize::try_from(semicolon_end).unwrap();
+            let semicolon_edit =
+                Edit::replacement(format!("\n{indentation}"), semicolon_start, semicolon_end);
+            return some_vec!(
+                Diagnostic::from_node(IfStatementSemicolon {}, &semicolon_node)
+                    .with_fix(Fix::safe_edits(if_edit, [semicolon_edit]))
+            );
+        }
+        None
+    }
+
+    fn entrypoints() -> Vec<&'static str> {
+        vec!["if_statement"]
+    }
+}
+
+/// Given an if statement node, convert it from a one-line if statement
+/// to a block if statement.
+/// This is a no-op if the node is already a block if statement.
+pub fn ifthenify(node: &Node, src: &SourceFile) -> Result<String> {
+    let source_text = src.source_text();
+    // check that the node is an if statement without an end if statement
+    if node
+        .named_descendants()
+        .any(|n| n.kind() == "end_if_statement")
+    {
+        return Ok(source_text.to_string());
+    }
+    // Divide the if statement into everything up to the end of the condition
+    // and the body.
+    let condition_node = node
+        .child_with_name("parenthesized_expression")
+        .context("If statement missing conditional")?;
+    let condition_end_byte = condition_node.end_byte();
+    // Concatenate bytes to the end of the condition, " then\n", the body, and
+    // "\nend if". Take care to get the indentation right.
+    let if_start_byte = node.start_byte();
+    let if_end_byte = node.end_byte();
+    let indentation = node.indentation(src);
+    let mut correction = String::new();
+    correction.push_str(&source_text[if_start_byte..condition_end_byte]);
+    correction.push_str(" then\n");
+    correction.push_str(&indentation);
+    correction.push_str("  "); // Assume two-space indent for the if body
```

Comment:
I should probably adjust this to account for whitespace between the end of the condition and the start of the body.

Reply:
Corrected in https://github.com/PlasmaFAIR/fortitude/pull/426/commits/dcf48cc92ecfc386551edd3425f750a6a2cbea09